src/MoBi.Presentation/DTO/CreateModuleDTO.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using MoBi.Assets;
+using OSPSuite.Assets;
+using OSPSuite.Presentation.DTO;
+using OSPSuite.Utility.Validation;
+
+namespace MoBi.Presentation.DTO
+{
+   public class CreateModuleDTO : ValidatableDTO
+   {
+      private IReadOnlyList<string> _prohibitedNames;
+
+      public CreateModuleDTO()
+      {
+         Rules.Add(ModuleCreationRules.UniqueName);
+         Rules.Add(ModuleCreationRules.NotEmptyName);
+      }
+
+      public void AddForbiddenNames(IReadOnlyList<string> prohibitedNames)
+      {
+         _prohibitedNames = prohibitedNames;
+      }
+
+      public string Name { get; set; }
+
+      public bool WithReaction { get; set; }
+      public bool WithEventGroup { get; set; }
+      public bool WithSpatialStructure { get; set; }
+      public bool WithPassiveTransport { get; set; }
+      public bool WithMolecule { get; set; }
+      public bool WithObserver { get; set; }
+      public bool WithMoleculeStartValues { get; set; }
+      public bool WithParameterStartValues { get; set; }
+
+      private bool isNameUnique(string newName)
+      {
+         if (_prohibitedNames == null)
+            return true;
+
+         return !_prohibitedNames.Contains(newName.Trim());
+      }
+      
+      private static class ModuleCreationRules
+      {
+         public static IBusinessRule NotEmptyName { get; } = CreateRule.For<CreateModuleDTO>()
+            .Property(item => item.Name)
+            .WithRule((dto, newName) => !string.IsNullOrEmpty(newName))
+            .WithError((dto, newName) => AppConstants.Validation.ModuleNameCannotBeEmpty);
+         
+         public static IBusinessRule UniqueName { get; } = CreateRule.For<CreateModuleDTO>()
+            .Property(item => item.Name)
```

Comment:
This one too probably

Reply:
I found one for non-blank name, but I don't see this one as a generic rule anywhere. Do you know where it might be?

Reply:
@rwmcintosh Hum not sure. Maybe it's in MoBi directly. We always have this concept of unique names etc.. Is there a BuidlgnBlockDTO. I think it's ok to duplicate but it's probably already there 5 times

Reply:
There are a couple of sort of generic ones. Like if you have an objectBaseDTO you can use the generic ones for ObjectBaseDTO. But this DTO doesn't have anything in common with ObjectBaseDTO. It's just a name and some bool properties. I'll put a generic rule in Core with the next commit.